src/NHibernate/Cache/ICacheConcurrencyStrategy.cs

```diff
@@ -153,11 +158,11 @@ internal static partial class CacheConcurrencyStrategyExtensions
 		/// <exception cref="CacheException"></exception>
 		public static object[] GetMany(this ICacheConcurrencyStrategy cache, CacheKey[] keys, long timestamp)
 		{
-			if (!(cache is IBatchableCacheConcurrencyStrategy batchableCache))
-			{
-				throw new InvalidOperationException($"Cache concurrency strategy {cache.GetType()} does not support batching");
-			}
-			return batchableCache.GetMany(keys, timestamp);
+			// PreferMultipleGet yields false if !IBatchableCacheConcurrencyStrategy, no GetMany call should be done
+			// in such case.
+			return ReflectHelper
+				.CastOrThrow<IBatchableCacheConcurrencyStrategy>(cache, "batching")
```

Comment:
This `GetMany` is currently called only in case of lazy batch fetches (using `batch-size`). The cases where NHibernate calls `GetMany` even if `PreferMultipleGet` is false do not use this `GetMany`.